spec/posix.test.sh

```diff
@@ -142,3 +142,41 @@ three
 EOF2
 ## stdout-json: "one\ntwo\nthree\n"
 
+#### source works for files in subdirectory
```

Comment:
Since  you mention this  isn't a POSIX feature, please move the tests to spec/builtin-source-eval.

The way I find tests for a particular feature:

`grep source spec/*.sh`

Reply:
I moved the tests, could you take another look at this?